week-2-practise/level-3/3-sort.js

```diff
@@ -4,12 +4,29 @@
 
 var nums = [10, 1, 5, 29, 100];
 var nums2 = [11, 6, 3, 29, 12];
-var sortedNums; // complete this statement
+var sortedNums = nums.concat(nums2).sort(function(a, b) {
+  return a - b;
+}); // complete this statement
+//if we don't use the compare function the numbers will convert the numbers to strings and sorts them
+// according to that ,which will lead to mistakes like  100 less than 29 and the result
+// will be like this [1,10,100,11,12,29,29,3,5,6]
 
 console.log(sortedNums);
 
 // 2. Using code, show that the variables nums and nums2 were not changed
 
+function isNumber(element) {
+  return typeof element === "number";
+}
+console.log(
+  " Did we change the variable type in the arrays nums and nums2 ? " +
+    !sortedNums.every(isNumber)
+);
+/* I tried the following function but it didn't work ,it gave me an error that the function Array.observe is not A function 
+can you tell me why ? 
```

Comment:
Check the warning message at the top of this page - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/observe - also the browser compatibility table lower down. Where did you find that `Array.observe()` was an option for this task?

Reply:
No  but he was asking  if there was any change happened to the array and array observe  one of the method which will .